sql/2020/05_Media/video_source_types.sql

```diff
@@ -0,0 +1,21 @@
+#standardSQL
+# get video source types
+# TODO: replace sample_data with the real data when done
+
+SELECT client,
+  video_source as video_type,
+  count(0) as video_type_count
+FROM
+  (
+  SELECT
+    client,
+    page,
+    video_source
+  FROM
+    `httparchive.sample_data.summary_response_bodies_firstHtml`,
+    UNNEST(regexp_extract_ALL(body, r'<video[^><]*>.*?<source[^><]*type=(?:"|\')*video/([^"|\'|\\s|>]*)(?:"|\'|\\s|>).*?</video>')) as video_source
+  )
+GROUP BY
+  client, video_source
+ORDER BY
+  client, video_type_count desc;
```

Comment:
I'm not convinced about this query. This will only pick up the first result, when a `<video>` element [can contain multiple sources](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source), such as:

```html
<video controls
    width="250"
    height="200"
    muted>
    <source src="/media/cc0-videos/flower.webm"
            type="video/webm">
    <source src="/media/cc0-videos/flower.mp4"
            type="video/mp4">
    This browser does not support the HTML5 video element.
</video>
```

What about below query to look at all the `types` (which doesn't have to be specified but should be to avoid a download for an unsupported type):

```sql
SELECT
  client,
  video_type,
  COUNT(0) AS video_type_count,
  SAFE_DIVIDE(COUNT(0), SUM(COUNT(0)) OVER (PARTITION BY client)) AS pct_video_types
FROM
    `httparchive.sample_data.summary_response_bodies_firstHtml`,
    #`httparchive.almanac.summary_response_bodies`,
    UNNEST(REGEXP_EXTRACT_ALL(body, '(?i)(<video.*?</video>)')) AS video,
    UNNEST(REGEXP_EXTRACT_ALL(video, r'(?i)type\s*=\s*["\'](video/[^\'";]*)')) AS video_type
WHERE
  date = '2020-08-01' AND
  firstHtml
GROUP BY
  client,
  video_type
ORDER BY
  client,
  video_type_count desc
```

This was basically stolen from [this query from 2019 media chapter](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/2019/04_Media/04_19.sql) but improved upon (at least I think so!)